.github/workflows/build-tool.yml

```diff
@@ -6,50 +6,98 @@ on:
         with:
           node-version: 16
 
-      - name: Build
-        shell: pwsh
-        env:
-          MORYX_BUILDNUMBER: ${{inputs.MORYX_BUILDNUMBER}}
-          MORYX_OPTIMIZE_CODE: ${{inputs.MORYX_OPTIMIZE_CODE}}
-          MORYX_BUILD_CONFIG: ${{inputs.MORYX_BUILD_CONFIG}}
-          DOTNET_SKIP_FIRST_TIME_EXPERIENCE: ${{inputs.DOTNET_SKIP_FIRST_TIME_EXPERIENCE}}
-        run: ./Build.ps1 -Build -Pack
+      - name: Setup .NET SDK standard (7)
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ inputs.dotnet_sdk_version }}
+
+      - name: Clean
+        run: dotnet clean && dotnet nuget locals all --clear
+
+      - name: Execute dotnet restore
+        run: dotnet restore
 
+      - name: Execute dotnet build
+        if: ${{ !(startsWith(github.ref, 'refs/tags/v')) }}
+        run: |
+          ASSEMBLY_VERSION="$(cat VERSION)"
+          ASSEMBLY_VERSION="${ASSEMBLY_VERSION%%.*}".0.0.0
+          echo "ASSEMBLY_VERSION = ${ASSEMBLY_VERSION}"
+          FILE_VERSION="$(cat VERSION)"."${{github.run_number}}"
+          echo "FILE_VERSION = ${FILE_VERSION}"
+          INFORMATIONAL_VERSION="$(cat VERSION)"-"${{github.ref_name}}"."${{github.run_number}}"
+          echo "INFORMATIONAL_VERSION = ${INFORMATIONAL_VERSION}"
+          dotnet build --no-restore -p:AssemblyVersion=$ASSEMBLY_VERSION -p:FileVersion=$FILE_VERSION -p:InformationalVersion=$INFORMATIONAL_VERSION
+        
+        
+      - name: Execute dotnet build (Release) 
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: |
+          ASSEMBLY_VERSION="${{github.ref_name}}"
+          ASSEMBLY_VERSION="${ASSEMBLY_VERSION#v}"
+          ASSEMBLY_VERSION="${ASSEMBLY_VERSION%%.*}".0.0.0
+          echo "ASSEMBLY_VERSION = ${ASSEMBLY_VERSION}"
+          FILE_VERSION="$(cat VERSION)"."${{github.run_number}}"
+          echo "FILE_VERSION = ${FILE_VERSION}"
+          INFORMATIONAL_VERSION="$(cat VERSION)"-"${{github.ref_name}}"."${{github.run_number}}"
+          echo "INFORMATIONAL_VERSION = ${INFORMATIONAL_VERSION}"
+          dotnet build --no-restore -p:AssemblyVersion=$ASSEMBLY_VERSION -p:FileVersion=$FILE_VERSION -p:InformationalVersion=$INFORMATIONAL_VERSION
+        
+        
+      - name: Execute dotnet pack with release version
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: |
+          RELEASE_VERSION="${{github.ref_name}}"
+          RELEASE_VERSION="${RELEASE_VERSION#v}"
+          echo "final RELEASE_VERSION = ${RELEASE_VERSION}"
+          dotnet pack --no-build --no-restore --output artifacts/packages -p:PackageVersion=$RELEASE_VERSION
+         
+        
+      - name: Execute dotnet pack with production version (dev or future)
+        if: ${{ github.ref_name == 'dev' || github.ref_name == 'future'}}
+        run: |
+          PRODUCTION_VERSION="$(cat VERSION)"-"${{github.ref_name}}"."${{github.run_number}}"
+          echo "final PRODUCTION_VERSION = ${PRODUCTION_VERSION}"
```

Comment:
Could you remove the final here? It feels like an artifact from your iterative process of concatinating the production version 😅

Reply:
Done